subsys/bluetooth/host/scan.c

```diff
@@ -31,6 +31,11 @@ static sys_slist_t scan_cbs = SYS_SLIST_STATIC_INIT(&scan_cbs);
 static struct bt_le_per_adv_sync *get_pending_per_adv_sync(void);
 static struct bt_le_per_adv_sync per_adv_sync_pool[CONFIG_BT_PER_ADV_SYNC_MAX];
 static sys_slist_t pa_sync_cbs = SYS_SLIST_STATIC_INIT(&pa_sync_cbs);
+#if CONFIG_BT_PER_ADV_SYNC_BUF_SIZE > 0
+/* A buffer pool used to reassemble periodic advertising data from the controller. */
+NET_BUF_POOL_FIXED_DEFINE(per_adv_sync_buf_pool, CONFIG_BT_PER_ADV_SYNC_MAX,
+			  CONFIG_BT_PER_ADV_SYNC_BUF_SIZE, NULL);
+#endif /* CONFIG_BT_PER_ADV_SYNC_BUF_SIZE > 0 */
```

Comment:
If there is a 1:1 between the reassembly buffers and the periodic advertising sync objects (which it seems there is), we could simply define this buffer in the struct, rather than having to allocate a pool and request the buffer from that pool. A single periodic sync object cannot use multiple buffers, so the pool design does not provide anything additional compared to defining the buffer inside the struct itself.

Reply:
After looking at the buffer documentation (https://docs.zephyrproject.org/latest/reference/networking/net_buf.html) I couldn't really find a way to define a network buffer in the struct.

Can you elaborate the approach you had in mind?

Reply:
@bewi-nordic You simply have to declare an array (of size `CONFIG_BT_PER_ADV_SYNC_BUF_SIZE` and a `struct net_buf buf;` in the struct. You will have to do the initialization manually though. Consider if a `struct net_buf_simple` is sufficient for the use case.